catalogue_graph/src/elasticsearch_transformers/concepts_transformer.py

```diff
@@ -0,0 +1,48 @@
+from models.indexable_concept import (
+    ConceptDisplay,
+    ConceptQuery,
+    IndexableConcept,
+    RelatedConcepts,
+)
+
+from .raw_neptune_concept import RawNeptuneConcept
+from .raw_neptune_related_concepts import RawNeptuneRelatedConcepts
+
+
+class ElasticsearchConceptsTransformer:
+    def transform_document(
+        self, concept: dict, related_concepts: dict
+    ) -> IndexableConcept:
+        neptune_concept = RawNeptuneConcept(concept, related_concepts)
```

Comment:
I think it would be better to push the responsibility for ensuring `neptune_concept` and `neptune_related` are of the correct type to the calling function.